builtin/credential/userpass/path_login.go

```diff
@@ -62,32 +63,56 @@ func (b *backend) pathLogin(ctx context.Context, req *logical.Request, d *framew
 	}
 
 	// Get the user and validate auth
-	user, err := b.user(ctx, req.Storage, username)
-	if err != nil {
-		return nil, err
-	}
-	if user == nil {
-		return logical.ErrorResponse("invalid username or password"), nil
-	}
-
-	// Check for a CIDR match.
-	if !cidrutil.RemoteAddrIsOk(req.Connection.RemoteAddr, user.BoundCIDRs) {
-		return logical.ErrorResponse("login request originated from invalid CIDR"), nil
+	user, userError := b.user(ctx, req.Storage, username)
+
+	var userPassword []byte
+	var err error
+	var legacyPassword bool
+	// If there was an error or it's nil, we fake a password for the bcrypt
+	// check so as not to have a timing leak. Specifics of the underlying
+	// storage still leaks a bit but generally much more in the noise compared
```

Comment:
```suggestion
	// storage still leaks a bit but generally much less in the noise compared
```

Reply:
It's more "in the noise" :-)

Reply:
I interpreted it as the storage call generating much more noise (for the leak to be easily observable) compared to the bcrypt call.